*** Makefile.in.orig	Wed May 25 17:31:24 2005
--- Makefile.in	Wed May 25 17:30:36 2005
***************
*** 404,410 ****
  # where is gmp?
  #---------------------------------------------
  # export gmpdir := $(PWD)/../gmp-4.1
! export gmpdir := /usr/local
  #---------------------------------------------
  # where is gmp lib?
  export gmplibdir = $(gmpdir)/lib
--- 404,410 ----
  # where is gmp?
  #---------------------------------------------
  # export gmpdir := $(PWD)/../gmp-4.1
! export gmpdir := $(PWD)/../../../..
  #---------------------------------------------
  # where is gmp lib?
  export gmplibdir = $(gmpdir)/lib
