diff -ubBr Makefile.in.orig Makefile.in
--- Makefile.in.orig	2011-06-07 22:56:05.000000000 +0900
+++ Makefile.in	2021-03-23 14:57:31.896967000 +0900
@@ -5,7 +5,7 @@
 
 @SET_MAKE@
 
-ALL_SUBDIRS=	cc
+ALL_SUBDIRS=	cpp
 DIST_SUBDIRS=	$(ALL_SUBDIRS) f77
 
 all install clean:
diff -ubBr configure.orig configure
--- configure.orig	2020-06-13 23:55:53.000000000 +0900
+++ configure	2021-03-23 15:00:01.478465000 +0900
@@ -2598,6 +2598,7 @@
 	    i?86) targmach=i386 ;;
 	    powerpc) targmach=powerpc endian=big ;;
 	    x86_64) targmach=amd64 ;;
+	    arm*) targmach=arm ;;
 	esac
         ;;
 
@@ -5445,7 +5446,7 @@
 _ACEOF
 
 
-ac_config_files="$ac_config_files Makefile cc/Makefile cc/cc/Makefile cc/cpp/Makefile cc/ccom/Makefile cc/cxxcom/Makefile cc/driver/Makefile f77/Makefile f77/f77/Makefile f77/fcom/Makefile"
+ac_config_files="$ac_config_files Makefile cpp/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -6143,7 +6143,7 @@
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
     "cc/Makefile") CONFIG_FILES="$CONFIG_FILES cc/Makefile" ;;
     "cc/cc/Makefile") CONFIG_FILES="$CONFIG_FILES cc/cc/Makefile" ;;
-    "cc/cpp/Makefile") CONFIG_FILES="$CONFIG_FILES cc/cpp/Makefile" ;;
+    "cpp/Makefile") CONFIG_FILES="$CONFIG_FILES cpp/Makefile" ;;
     "cc/ccom/Makefile") CONFIG_FILES="$CONFIG_FILES cc/ccom/Makefile" ;;
     "cc/cxxcom/Makefile") CONFIG_FILES="$CONFIG_FILES cc/cxxcom/Makefile" ;;
     "cc/driver/Makefile") CONFIG_FILES="$CONFIG_FILES cc/driver/Makefile" ;;
diff -ubBr cc/cpp/Makefile.in.orig cpp/Makefile.in
--- cc/cpp/Makefile.in.orig	2020-06-29 22:37:18.000000000 +0900
+++ cc/cpp/Makefile.in	2021-03-23 15:04:06.518851000 +0900
@@ -32,7 +32,7 @@
 MDIR=$(top_srcdir)/arch/$(TARGMACH)
 COMMONDIR=$(top_srcdir)/common
 
-DEST=@BINPREFIX@cpp$(EXEEXT)
+DEST=@BINPREFIX@ox_cpp$(EXEEXT)
 MANPAGE=@BINPREFIX@cpp
 
 all: $(DEST)
diff -ubBr cc/cpp/cpp.c.orig cc/cpp/cpp.c
--- cc/cpp/cpp.c.orig	2020-02-27 02:58:19.000000000 +0900
+++ cc/cpp/cpp.c	2021-03-25 17:01:14.157700000 +0900
@@ -802,10 +802,8 @@
 	free(nm);
 	nm = addname(ob->buf);
 	bufree(ob);
-	if (pushfile(nm, fn, SYSINC, NULL) == 0)
+	pushfile(nm, fn, SYSINC, NULL);
 		return 1;
-
-	return 0;
 }
 
 #endif
