diff -urN gnuplot-4.0.0.orig/configure.in gnuplot-4.0.0/configure.in
--- gnuplot-4.0.0.orig/configure.in	Sun Apr 11 00:55:14 2004
+++ gnuplot-4.0.0/configure.in	Wed Dec 15 11:07:49 2004
@@ -827,6 +827,10 @@
            tutorial/Makefile],
           [test -z "$CONFIG_HEADERS" || echo timestamp >stamp-h])
 
+dnl for Solaris 2.x
+AC_CHECK_FUNC(socket, , [AC_CHECK_LIB(socket,socket)])
+AC_CHECK_FUNC(gethostbyname, , [AC_CHECK_LIB(nsl,gethostbyname)])
+
 dnl Report configuration
 AC_MSG_RESULT([
 ** Configuration summary for $PACKAGE $VERSION:
