diff -ubBr pari-2.0.17.beta.orig/Configure pari-2.0.17.beta/Configure
--- pari-2.0.17.beta.orig/Configure	Tue Sep 21 00:53:58 1999
+++ pari-2.0.17.beta/Configure	Tue Dec  7 23:27:09 2004
@@ -859,7 +859,7 @@
             cppfl="-DUSE_ELF"    
     fi ;;
 esac
-CPP="$cc $cppfl -E -I."
+CPP="$CC $cppfl -E -I."
 
 echo "C PreProcessor is      $CPP"
 
diff -ubBr pari-2.0.17.beta.orig/Odos/paricfg.h pari-2.0.17.beta/Odos/paricfg.h
--- pari-2.0.17.beta.orig/Odos/paricfg.h	Tue Sep 21 00:56:06 1999
+++ pari-2.0.17.beta/Odos/paricfg.h	Tue Dec  7 23:27:09 2004
@@ -2,7 +2,7 @@
 #ifndef __CONFIG_H__
 #define __CONFIG_H__
 
-#ifdef _MSC_VER /* Bill Daly: avoid spurious Warnings from MSVS */
+#ifdef _MSC_VER /* Bill Daly: avoid spurious Warnings from MSVC */
 #  pragma warning(disable: 4013 4018 4146 4244 4761)
 #endif
 
diff -ubBr pari-2.0.17.beta.orig/config/Makefile.SH pari-2.0.17.beta/config/Makefile.SH
--- pari-2.0.17.beta.orig/config/Makefile.SH	Thu Sep 16 22:48:59 1999
+++ pari-2.0.17.beta/config/Makefile.SH	Tue Dec  7 23:29:18 2004
@@ -429,17 +429,17 @@
   plot-dyn)
     source="$src/graph/\$(PLOTFILE)"
     cflags="\$(DYNFLAGS) \$(PLOTCFLAGS)"
-    depend="$src/graph/rect.h libpari.dll libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/graph/rect.h libpari.dll libpari_dll.h libpari.dll"
     ;;
   gp-dyn|gp_rl-dyn)
     source=$src\/`echo $dir | sed -e "s/dyn//"`\/`echo $f | cut -f1 -d-`.c
     cflags="\$(DYNFLAGS) -I$src/language \$(RLINCLUDE)"
-    depend="$src/language/anal.h $src/gp/gp.h ./paricfg.h libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/language/anal.h $src/gp/gp.h ./paricfg.h libpari_dll.h libpari.dll"
     ;;
   gp_init-dyn|plotport-dyn)
     source=$src\/`echo $dir | sed -e "s/dyn//"`\/`echo $f | cut -f1 -d-`.c
     cflags="\$(DYNFLAGS) -I$src/graph"
-    depend="$src/graph/rect.h libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/graph/rect.h libpari_dll.h libpari.dll"
     ;;
   init|es|sumiter)
     depend="$src/language/anal.h"
diff -ubBr pari-2.0.17.beta.orig/config/TOP_Make.SH pari-2.0.17.beta/config/TOP_Make.SH
--- pari-2.0.17.beta.orig/config/TOP_Make.SH	Thu Sep 16 22:48:59 1999
+++ pari-2.0.17.beta/config/TOP_Make.SH	Tue Dec  7 23:27:09 2004
@@ -48,7 +48,7 @@
 	@echo "	ctags			Generate VI/VIM tags file in ./src"
 	@echo "	etags			Generate Emacs  tags file in ./src"
 
-gp all bench test-compat test-graphic install clean cleantest install-bin install-doc install-lib-sta install-bin-sta dobench::
+gp all lib-sta bench test-compat test-graphic install clean cleantest install-bin install-doc install-lib-sta install-bin-sta install-lib-dyn install-data install-include install-man install-misc dobench::
 	cd $objdir; \$(MAKE) \$@
 
 doc docps gpman::
diff -ubBr pari-2.0.17.beta.orig/config/has_exp2.c pari-2.0.17.beta/config/has_exp2.c
--- pari-2.0.17.beta.orig/config/has_exp2.c	Thu Sep 16 22:48:59 1999
+++ pari-2.0.17.beta/config/has_exp2.c	Tue Dec  7 23:27:09 2004
@@ -1,2 +1,2 @@
 #include <math.h>
-main(){double x=exp2(1.0);}
+main(){double x=exp2(2.017);}
diff -ubBr pari-2.0.17.beta.orig/src/headers/paricom.h pari-2.0.17.beta/src/headers/paricom.h
--- pari-2.0.17.beta.orig/src/headers/paricom.h	Thu Sep 16 22:50:21 1999
+++ pari-2.0.17.beta/src/headers/paricom.h	Tue Dec  7 23:27:09 2004
@@ -86,7 +86,9 @@
 #else
   BEGINEXTERN
     double exp2(double);
+#    ifndef __CYGWIN32__
     double log2(double);
+#    endif
   ENDEXTERN
 #endif
 
