diff -ubBr pari-2.3.4.orig/src/desc/gen_member pari-2.3.4/src/desc/gen_member
--- pari-2.3.4.orig/src/desc/gen_member	2008-01-17 00:55:50.000000000 +0900
+++ pari-2.3.4/src/desc/gen_member	2020-01-16 12:36:42.888060000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/perl -w
+BEGIN { push @INC, "." if $INC[-1] ne "."; }
 use PARI::822;
 
 $class='member_functions';
diff -ubBr pari-2.3.4.orig/src/desc/gen_proto pari-2.3.4/src/desc/gen_proto
--- pari-2.3.4.orig/src/desc/gen_proto	2008-01-17 00:55:51.000000000 +0900
+++ pari-2.3.4/src/desc/gen_proto	2020-01-16 12:36:44.871411000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/perl -w
+BEGIN { push @INC, "." if $INC[-1] ne "."; }
 use PARI::822;
 
 $class=$ARGV[0];
diff -ubBr pari-2.3.4.orig/src/desc/merge_822 pari-2.3.4/src/desc/merge_822
--- pari-2.3.4.orig/src/desc/merge_822	2008-01-17 00:55:50.000000000 +0900
+++ pari-2.3.4/src/desc/merge_822	2020-01-16 12:36:57.223570000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/perl -w
+BEGIN { push @INC, "." if $INC[-1] ne "."; }
 use PARI::822;
 
 PARI::822::read(\%funcs,$_,1)
