diff -ubBr pari-2.2.8.alpha.orig/config/DOC_Make.SH pari-2.2.8.alpha/config/DOC_Make.SH
--- pari-2.2.8.alpha.orig/config/DOC_Make.SH	Thu Sep  2 08:48:37 2004
+++ pari-2.2.8.alpha/config/DOC_Make.SH	Fri Dec 10 03:31:23 2004
@@ -12,7 +12,7 @@
 docps: libpari.ps refcard.ps tutorial.ps users.ps INSTALL.ps
 docpdf: libpari.pdf users.pdf tutorial.pdf refcard.pdf INSTALL.pdf
 
-.SUFFIXES: .tex .ipf .pod .inf .dumbpod .3 .html
+.SUFFIXES: .tex .ipf .pod .inf .dumbpod .3 .html .idx .std .dvi .ps .pdf
 
 paricfg.tex: paricfg.tex.in
 	sed -e 's/@version@/$version.$patch/'\\
@@ -20,22 +20,22 @@
 	    -e 's!@libdir@!$libdir!'\\
 	    -e 's!@datadir@!$datadir!' paricfg.tex.in > \$@
 
-%.pdf: %.tex \$(MACROS)
+.tex.pdf: \$(MACROS)
 	pdftex \$<
 
-%.dvi: %.tex \$(MACROS)
+.tex.dvi: \$(MACROS)
 	tex \$<
 
 libpari.pdf: \$(PARI_TEX) \$(MACROS)
 	-rm -f libpari.std
 	pdftex libpari
 	pdftex libpari
-	make libpari.std; pdftex libpari; rm -f libpari.std
+	\$(MAKE) libpari.std; pdftex libpari; rm -f libpari.std
 libpari.dvi: \$(PARI_TEX) \$(MACROS)
 	-rm -f libpari.std
 	tex libpari
 	tex libpari
-	make libpari.std; tex libpari; rm -f libpari.std
+	\$(MAKE) libpari.std; tex libpari; rm -f libpari.std
 
 tutorial.pdf: tutorial.tex \$(MACROS)
 	-rm -f tutorial.std
@@ -46,13 +46,13 @@
 	tex tutorial
 	tex tutorial
 
-%.ps: %.dvi
+.dvi.ps:
 	dvips \$< -o \$@
 
 refcard.ps: refcard.dvi
 	dvips -t landscape -t a4 refcard.dvi -o \$@
 
-%.std: %.idx
+.idx.std:
 	sed -e 's/!\([1-9]\)!/!00\1!/'\\
 	    -e 's/!\([1-9][0-9]\)!/!0\1!/'\\
 	    -e 's/\\\\EFF {}/f/g'\\
@@ -63,13 +63,13 @@
 	-rm -f users.std
 	tex users
 	tex users
-	make users.std; tex users; rm -f users.std
+	\$(MAKE) users.std; tex users; rm -f users.std
 
 users.pdf: \$(USERS_TEX)
 	-rm -f users.std
 	pdftex users
 	pdftex users
-	make users.std; pdftex users; rm -f users.std
+	\$(MAKE) users.std; pdftex users; rm -f users.std
 
 gpman: gp.1
 	nroff -man gp.1 | unix2dos -ascii > gp.man
diff -ubBr pari-2.2.8.alpha.orig/config/TOP_Make.SH pari-2.2.8.alpha/config/TOP_Make.SH
--- pari-2.2.8.alpha.orig/config/TOP_Make.SH	Thu Jan 15 04:33:33 2004
+++ pari-2.2.8.alpha/config/TOP_Make.SH	Fri Dec 10 02:56:13 2004
@@ -41,7 +41,7 @@
 	@-cd doc && \$(MAKE) doc
 	@\$(MAKE) gp
 
-gp bench test-kernel test-all $top_test_extra $top_dotest_extra install cleanobj cleantest install-bin install-doc install-data install-lib-sta install-bin-sta dobench tune::
+gp bench test-kernel test-all $top_test_extra $top_dotest_extra install cleanobj cleantest install-bin install-doc install-data install-lib-sta install-bin-sta dobench tune lib-sta install-lib-dyn install-include install-man install-misc::
 	@dir=\`config/objdir\`; echo "Making \$@ in \$\$dir";\\
 	 if test ! -d \$\$dir; then echo "Please run Configure first!"; exit 1; fi;\\
 	cd \$\$dir && \$(MAKE) \$@
