diff -ubBr pari-2.0.21.beta.orig/Configure pari-2.0.21.beta/Configure
--- pari-2.0.21.beta.orig/Configure	Sat Oct 28 04:04:49 2000
+++ pari-2.0.21.beta/Configure	Wed Jul 13 13:25:47 2005
@@ -225,7 +225,8 @@
   irix*)    osname=irix;;
   fx2800)   arch=fx2800; osname=concentrix;;
   hp*)      arch=hppa; osname=hpux;;
-  freebsd|os2)  arch=ix86;;
+  freebsd)  arch=`uname -m`;;
+  os2)  arch=ix86;;
   ultrix)   arch=mips;;
   nextstep) arch=`file /bin/sh | sed 's/.*(for architecture \(.*\))/\1/'`;;
   osf1)     case "$5" in alpha) arch=alpha;; esac;;
diff -ubBr pari-2.0.21.beta.orig/config/Makefile.SH pari-2.0.21.beta/config/Makefile.SH
--- pari-2.0.21.beta.orig/config/Makefile.SH	Tue Mar 28 02:43:42 2000
+++ pari-2.0.21.beta/config/Makefile.SH	Fri Dec 10 20:13:19 2004
@@ -422,17 +422,17 @@
   plot-dyn)
     source="$src/graph/\$(PLOTFILE)"
     cflags="\$(DYNFLAGS) \$(PLOTCFLAGS)"
-    depend="$src/graph/rect.h libpari.dll libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/graph/rect.h libpari.dll libpari_dll.h libpari.dll"
     ;;
   gp-dyn|gp_rl-dyn)
     source=$src\/`echo $dir | sed -e "s/dyn//"`\/`echo $f | cut -f1 -d-`.c
     cflags="\$(DYNFLAGS) -I$src/language \$(RLINCLUDE)"
-    depend="$src/language/anal.h $src/gp/gp.h ./paricfg.h libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/language/anal.h $src/gp/gp.h ./paricfg.h libpari_dll.h libpari.dll"
     ;;
   gp_init-dyn|plotport-dyn)
     source=$src\/`echo $dir | sed -e "s/dyn//"`\/`echo $f | cut -f1 -d-`.c
     cflags="\$(DYNFLAGS) -I$src/graph"
-    depend="$src/graph/rect.h libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/graph/rect.h libpari_dll.h libpari.dll"
     ;;
   init|es|sumiter)
     depend="$src/language/anal.h"
diff -ubBr pari-2.0.21.beta.orig/config/TOP_Make.SH pari-2.0.21.beta/config/TOP_Make.SH
--- pari-2.0.21.beta.orig/config/TOP_Make.SH	Wed Jul 12 03:12:27 2000
+++ pari-2.0.21.beta/config/TOP_Make.SH	Fri Dec 10 20:15:54 2004
@@ -48,7 +48,7 @@
 	@echo "	ctags			Generate VI/VIM tags file in ./src"
 	@echo "	etags			Generate Emacs  tags file in ./src"
 
-gp all bench test-compat test-graphic install cleanobj cleantest install-bin install-doc install-lib-sta install-bin-sta dobench::
+gp all bench test-compat test-graphic install cleanobj cleantest install-bin install-doc install-lib-sta install-bin-sta dobench lib-sta install-lib-dyn install-data install-include install-man install-misc::
 	cd $objdir; \$(MAKE) \$@
 
 doc docps docpdf gpman cleandoc::
diff -ubBr pari-2.0.21.beta.orig/config/has_exp2.c pari-2.0.21.beta/config/has_exp2.c
--- pari-2.0.21.beta.orig/config/has_exp2.c	Thu Sep 16 22:46:59 1999
+++ pari-2.0.21.beta/config/has_exp2.c	Fri Dec 10 20:13:19 2004
@@ -1,2 +1,2 @@
 #include <math.h>
-main(){double x=exp2(1.0);}
+main(){double x=exp2(2.017);}
diff -ubBr pari-2.0.21.beta.orig/src/headers/paricom.h pari-2.0.21.beta/src/headers/paricom.h
--- pari-2.0.21.beta.orig/src/headers/paricom.h	Fri Oct 27 20:21:34 2000
+++ pari-2.0.21.beta/src/headers/paricom.h	Fri Dec 10 20:13:19 2004
@@ -83,7 +83,9 @@
 #else
   BEGINEXTERN
     double exp2(double);
+#    ifndef __CYGWIN32__
     double log2(double);
+#    endif
   ENDEXTERN
 #endif
 
diff -ubBr pari-2.0.21.beta.orig/src/kernel/ix86/level0.h pari-2.0.21.beta/src/kernel/ix86/level0.h
--- pari-2.0.21.beta.orig/src/kernel/ix86/level0.h	2000-09-15 19:52:10.000000000 +0900
+++ pari-2.0.21.beta/src/kernel/ix86/level0.h	2008-08-12 04:38:59.000000000 +0900
@@ -159,8 +159,8 @@
 #define divll(a,b) \
 ({ ulong __value, __arg1 = (a), __arg2 = (b); \
    __asm__ ("divl %4" \
-        : "=a" /* %eax */ (__value), "=d" /* %edx */ (hiremainder) \
-        : "0" /* %eax */ (__arg1), "1" /* %edx */ (hiremainder), "g" (__arg2)); \
+        : "=a" /* %eax */ (__value), "=&d" /* %edx */ (hiremainder) \
+        : "0" /* %eax */ (__arg1), "1" /* %edx */ (hiremainder), "mr" (__arg2)); \
    __value; \
 })
 
