diff -ubBr pari-2.0.21.beta.orig/config/Makefile.SH pari-2.0.21.beta/config/Makefile.SH
--- pari-2.0.21.beta.orig/config/Makefile.SH	Tue Mar 28 02:43:42 2000
+++ pari-2.0.21.beta/config/Makefile.SH	Fri Dec 10 20:13:19 2004
@@ -422,17 +422,17 @@
   plot-dyn)
     source="$src/graph/\$(PLOTFILE)"
     cflags="\$(DYNFLAGS) \$(PLOTCFLAGS)"
-    depend="$src/graph/rect.h libpari.dll libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/graph/rect.h libpari.dll libpari_dll.h libpari.dll"
     ;;
   gp-dyn|gp_rl-dyn)
     source=$src\/`echo $dir | sed -e "s/dyn//"`\/`echo $f | cut -f1 -d-`.c
     cflags="\$(DYNFLAGS) -I$src/language \$(RLINCLUDE)"
-    depend="$src/language/anal.h $src/gp/gp.h ./paricfg.h libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/language/anal.h $src/gp/gp.h ./paricfg.h libpari_dll.h libpari.dll"
     ;;
   gp_init-dyn|plotport-dyn)
     source=$src\/`echo $dir | sed -e "s/dyn//"`\/`echo $f | cut -f1 -d-`.c
     cflags="\$(DYNFLAGS) -I$src/graph"
-    depend="$src/graph/rect.h libpari_dll.h libpari_globals.h libpari.dll"
+    depend="$src/graph/rect.h libpari_dll.h libpari.dll"
     ;;
   init|es|sumiter)
     depend="$src/language/anal.h"
diff -ubBr pari-2.0.21.beta.orig/config/TOP_Make.SH pari-2.0.21.beta/config/TOP_Make.SH
--- pari-2.0.21.beta.orig/config/TOP_Make.SH	Wed Jul 12 03:12:27 2000
+++ pari-2.0.21.beta/config/TOP_Make.SH	Fri Dec 10 20:15:54 2004
@@ -48,7 +48,7 @@
 	@echo "	ctags			Generate VI/VIM tags file in ./src"
 	@echo "	etags			Generate Emacs  tags file in ./src"
 
-gp all bench test-compat test-graphic install cleanobj cleantest install-bin install-doc install-lib-sta install-bin-sta dobench::
+gp all bench test-compat test-graphic install cleanobj cleantest install-bin install-doc install-lib-sta install-bin-sta dobench lib-sta install-lib-dyn install-data install-include install-man install-misc::
 	cd $objdir; \$(MAKE) \$@
 
 doc docps docpdf gpman cleandoc::
diff -ubBr pari-2.0.21.beta.orig/config/has_exp2.c pari-2.0.21.beta/config/has_exp2.c
--- pari-2.0.21.beta.orig/config/has_exp2.c	Thu Sep 16 22:46:59 1999
+++ pari-2.0.21.beta/config/has_exp2.c	Fri Dec 10 20:13:19 2004
@@ -1,2 +1,2 @@
 #include <math.h>
-main(){double x=exp2(1.0);}
+main(){double x=exp2(2.017);}
diff -ubBr pari-2.0.21.beta.orig/src/headers/paricom.h pari-2.0.21.beta/src/headers/paricom.h
--- pari-2.0.21.beta.orig/src/headers/paricom.h	Fri Oct 27 20:21:34 2000
+++ pari-2.0.21.beta/src/headers/paricom.h	Fri Dec 10 20:13:19 2004
@@ -83,7 +83,9 @@
 #else
   BEGINEXTERN
     double exp2(double);
+#    ifndef __CYGWIN32__
     double log2(double);
+#    endif
   ENDEXTERN
 #endif
 
